### PR TITLE
LCAM-1191

### PR DESCRIPTION
**Pull request by skasthuri36326**
## What

[LCAM-1191](https://dsdmoj.atlassian.net/browse/LCAM-1191)

Describe what you did and why.
- Migrated validation around Crown Court Outcomes in the pre_update_checks Stored Procedure to the Crown Court Proceeding Service

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] GitHub should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

[LCAM-1191]: https://dsdmoj.atlassian.net/browse/LCAM-1191?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/validation/CrownCourtDetailsValidator.java [hunk 1]
```diff
@@ -2,40 +2,49 @@
...
-        if (apiCrownCourtOutcome.isPresent()) {
-            final CrownCourtOutcome crownCourtOutcome = apiCrownCourtOutcome.get().getOutcome();
-            log.info("Validating crown court outcome " + crownCourtOutcome);
-            if ((CONVICTED.getCode().equals(crownCourtOutcome.getCode()) || PART_CONVICTED.getCode().equals(crownCourtOutcome.getCode()))
-                    && crownCourtSummary.getIsImprisoned() == null) {
-                throw new ValidationException(MSG_INVALID_CC_OUTCOME);
+    public Optional<Void> checkCCDetails(CrownCourtDTO dto) {
+        ApiCrownCourtSummary crownCourtSummary = dto.getCrownCourtSummary();
+        List<RepOrderCCOutcomeDTO> repOrderCCOutcomeDTOList = maatCourtDataService.getRepOrderCCOutcomeByRepId(dto.getRepId(), null);
+        String caseType = dto.getCaseType() != null ? dto.getCaseType().getCaseType() : "X";
```
Comment (venkat980) [inline crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/validation/CrownCourtDetailsValidator.java:32]: I think this logic was recreated from the stored procedure, we dont need this. Instead, we can check if case type is not APPEAL_CC as below:
!CaseType.APPEAL_CC.getCaseType().equals(dto.getCaseType())

File: crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownCourtProceedingControllerTest.java [hunk 2]
```diff
@@ -145,4 +150,15 @@ void givenAValidInput_whenUpdateIsInvoked_RequestObjectFailsValidation() throws
                         HttpMethod.PUT, updateApplicationRequestJson, ENDPOINT_URL + "/update-crown-court"))
                 .andExpect(status().is4xxClientError());
     }
+
+    @Test
+    void givenAnInput_whenUpdateIsInvokedAndValidationFails_thenBadRequestResponseIsReturned() throws Exception {
+        var apiUpdateApplicationRequest =
```
Comment (venkat980) [inline crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownCourtProceedingControllerTest.java:156]: Just checking if the variable declaration was intentional here (apiUpdateApplicationRequest as var) This is always going to be ApiUpdateApplicationRequest.

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/validation/CrownCourtDetailsValidator.java (hunk 1 above)
Comment (skasthuri36326) [reply crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/validation/CrownCourtDetailsValidator.java:32]: Agree, caseType is a required field.

File: crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownCourtProceedingControllerTest.java (hunk 2 above)
Comment (skasthuri36326) [reply crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownCourtProceedingControllerTest.java:156]: No intention. Just to keep it consistent with the other test cases in CrownCourtProceedingControllerTest.java

**Review by venkat980 — APPROVED** [review]

**Pull request merged**